includes/class-wc-comments.php

```diff
@@ -337,11 +337,6 @@ public static function get_average_rating_for_product( &$product ) {
 			$average = 0;
 		}
 
-		$product->set_average_rating( $average );
-
-		$data_store = $product->get_data_store();
-		$data_store->update_average_rating( $product );
-
```

Comment:
I guess the product is no longer being modified here, so perhaps would be good to change the method signatures for all 3 functions.

Reply:
Isn't now the `set_average_rating` and `update_visibility` (called from `update_average_rating `) missing from `sync_average_rating`?

Reply:
Not sure I follow. `sync_average_rating` does a direct meta update or the average rating it's setting?

Reply:
What I meant is that previously, `sync_average_rating` included (via `get_average_rating_for_product`)
1. `set_average_rating` on `$product`
2. `update_post_meta` for meta_key `_wc_average_rating`
3. `WC_Product_Data_Store_CPT::update_visibility` on `$product`

Now, it only runs `update_post_meta` in the function `sync_average_rating` itself. 

So I thought that `update_visibility` has a potential side effect in the db, and it depends on `$product->get_average_rating()`, thus steps 1 and 3 might be needed in `sync_average_rating`?